Framework/ICat/src/CatalogDownloadDataFiles.cpp

```diff
@@ -169,6 +169,8 @@ std::string CatalogDownloadDataFiles::doDownloadandSavetoLocalDrive(
     Poco::Net::SecureStreamSocket *socket =
         new Poco::Net::SecureStreamSocket(context);
     Poco::Net::HTTPSClientSession session(*socket);
```

Comment:
The lines at the bottom of he function describe (from the original author Jay Rainey)

  // In addition, there's a crash when destructing SecureSocketImpl (internal to
  // SecureSocketStream, which is
  // created and destroyed by HTTPSClientSession). We avoid that crash by
  // instantiating SecureSocketStream
  // ourselves and passing it to the HTTPSClientSession, which takes ownership.

Martyn can remember that Jay tried it on the stack first, but had to move to the heap.  This could be reassesed once we have moved all env over to poco 1.6, but while we are on poco 1.4 it is not worth doing.